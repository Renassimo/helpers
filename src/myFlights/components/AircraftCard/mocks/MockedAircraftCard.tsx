const MockedAircraftCard = jest.fn(({ chosen }) => (
  <div>MockedAircraftCard - {chosen && 'chosen'}</div>
));

export default MockedAircraftCard;
