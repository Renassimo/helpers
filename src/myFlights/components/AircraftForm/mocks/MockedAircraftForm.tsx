const MockedAircraftForm = jest.fn(({ registration }) => (
  <div>MockedAircraftForm - {registration}</div>
));

export default MockedAircraftForm;
