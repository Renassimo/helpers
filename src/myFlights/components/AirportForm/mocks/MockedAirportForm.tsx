const MockedAirportForm = jest.fn(({ title }) => (
  <div>MockedAirportForm - {title}</div>
));

export default MockedAirportForm;
