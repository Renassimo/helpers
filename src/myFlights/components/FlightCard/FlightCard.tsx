import { Avia } from '@/avia/types/avia';

import AviaCard from '@/avia/components/AviaCard';

const FlightCard = ({
  data,
  onClick,
  chosen = false,
}: {
  data: Avia.FlightData;
  onClick: () => void;
  chosen?: boolean;
}) => {
  const { attributes } = data;
  const {
    registration,
    distance,
    airline,
    aircraft,
    origin,
    destination,
    originName,
    destinationName,
    photoUrl,
    flightNumber,
  } = attributes;

  const withRoute = origin || destination;
  const withNamedRoute = originName || destinationName;

  const additionalContent = [
    withNamedRoute && `${originName} - ${destinationName}`,
    withRoute && `${origin} - ${destination}`,
    airline,
    aircraft,
    registration,
    distance ? `${distance} km` : null,
  ];

  const link =
    flightNumber && `https://www.google.com/search?q=${flightNumber}`;

  return (
    <AviaCard
      onClick={onClick}
      chosen={chosen}
      imageUrl={photoUrl ?? '/images/plane.png'}
      imageAlt={flightNumber}
      link={link}
      title={flightNumber}
      additionalContent={additionalContent}
    />
  );
};

export default FlightCard;
