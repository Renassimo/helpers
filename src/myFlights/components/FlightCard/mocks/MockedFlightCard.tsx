const MockedFlightCard = jest.fn(({ data }) => (
  <div>MockedFlightCard - {data?.id}</div>
));

export default MockedFlightCard;
