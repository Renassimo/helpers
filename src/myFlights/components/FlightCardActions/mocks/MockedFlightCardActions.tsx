const MockedFlightCardActions = jest.fn(({ data }) => (
  <div>MockedFlightCardActions - {data?.id}</div>
));

export default MockedFlightCardActions;
