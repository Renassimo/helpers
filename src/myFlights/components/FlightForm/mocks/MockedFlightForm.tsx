const MockedFlightForm = jest.fn(() => <div>MockedFlightForm</div>);

export default MockedFlightForm;
