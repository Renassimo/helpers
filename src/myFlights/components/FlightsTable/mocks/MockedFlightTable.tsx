const MockedFlightTable = jest.fn(() => <div>MockedFlightTable</div>);

export default MockedFlightTable;
