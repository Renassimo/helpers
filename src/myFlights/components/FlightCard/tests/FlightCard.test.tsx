import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import AviaCard from '@/avia/components/AviaCard';

import MockedAviaCard from '@/avia/components/AviaCard/mocks';

import { mockedDeserializedFlights } from '@/avia/types/avia/mocks';

import FlightCard from '../FlightCard';

jest.mock('@/avia/components/AviaCard');

describe('FlightCard', () => {
  beforeEach(() => {
    (AviaCard as unknown as jest.Mock).mockImplementation(MockedAviaCard);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <FlightCard
        data={mockedDeserializedFlights[0]}
        chosen
        onClick={jest.fn()}
      />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when partial flight passed', () => {
    test('renders successfully', () => {
      // Arange
      const mockedFlight = {
        ...mockedDeserializedFlights[0],
        attributes: {
          ...mockedDeserializedFlights[0].attributes,
          origin: '',
          destinationName: '',
          distance: null,
        },
      };
      // Act
      const { baseElement } = renderWithTheme(
        <FlightCard data={mockedFlight} onClick={jest.fn()} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });
});
