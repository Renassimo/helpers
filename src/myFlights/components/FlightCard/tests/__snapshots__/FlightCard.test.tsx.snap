// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCard renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedAviaCard - 
      FZ 1839
       -
      <div>
        Dubai - Warsaw Chopin
      </div>
      <div>
        DXB/OMDB - WAW/EPWA
      </div>
      <div>
        flydubai
      </div>
      <div>
        Boeing 737 MAX 8
      </div>
      <div>
        A6-FMB
      </div>
      <div>
        4157 km
      </div>
       
      - 
      chosen
       - 
      https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg
       - 
      FZ 1839
    </div>
  </div>
</body>
`;

exports[`FlightCard when partial flight passed renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedAviaCard - 
      FZ 1839
       -
      <div>
        Dubai - 
      </div>
      <div>
         - WAW/EPWA
      </div>
      <div>
        flydubai
      </div>
      <div>
        Boeing 737 MAX 8
      </div>
      <div>
        A6-FMB
      </div>
       
      - 
       - 
      https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg
       - 
      FZ 1839
    </div>
  </div>
</body>
`;
